test/tests/functional/pbs_equiv_classes.py

```diff
@@ -1606,3 +1606,312 @@ class instead of the class it was in when it was queued.
+                                 max_attempts=10, starttime=self.t)
+
+        # create two new jobs for same equivalence classes
+        a = {'Resource_List.ncpus': 1}
+        J = Job(TEST_USER, attrs=a)
+        jid3 = self.server.submit(J)
+
+        a = {'Resource_List.ncpus': 1, 'Resource_List.walltime': 600}
+        J2 = Job(TEST_USER, attrs=a)
+        jid4 = self.server.submit(J2)
+
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid2)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid3)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid4)
+
+        # Suspend jobs
+        self.server.sigjob(jobid=jid1, signal="suspend")
+        self.server.sigjob(jobid=jid2, signal="suspend")
+
+        self.server.expect(JOB, {'job_state': 'S'}, id=jid1)
+        self.server.expect(JOB, {'job_state': 'S'}, id=jid2)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid3)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid4)
+
+        # Check that both suspended job is put in its own equivalence class
+        self.scheduler.log_match("Number of job equivalence classes: 4",
+                                 max_attempts=10, starttime=self.t)
+
+        a = {'Resource_List.ncpus': 1, 'queue': 'expressq'}
+        Je = Job(TEST_USER, attrs=a)
+        jid5 = self.server.submit(Je)
+        jid6 = self.server.submit(Je)
+
+        self.server.expect(JOB, {'job_state': 'S'}, id=jid1)
+        self.server.expect(JOB, {'job_state': 'S'}, id=jid2)
+        self.server.expect(JOB, {'job_state': 'S'}, id=jid3)
+        self.server.expect(JOB, {'job_state': 'S'}, id=jid4)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid5)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid6)
+
+        """Test only 3 equivalence class are present, i.e one
+         equivalence class for jid1 and jid3,1 equi class for jid2
+          and jid4 and 1 equi class for jid5 and jid6 """
+
+        self.scheduler.log_match("Number of job equivalence classes: 3",
+                                 max_attempts=10, starttime=self.t)
+
+    def test_held_jobs_equi_class(self):
+        "Test that held jobs do not go into another equivalence class."
+
+        a = {'resources_available.ncpus': 1}
+        self.server.create_vnodes('vnode', a, 2, self.mom, usenatvnode=True)
+
+        a = {'Resource_List.ncpus': 1}
+        J = Job(TEST_USER, attrs=a)
+
+        jid1 = self.server.submit(J)
+        jid2 = self.server.submit(J)
+        jid3 = self.server.submit(J)
+
+        self.server.holdjob(jid1, USER_HOLD)
+
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid3)
+
+        self.scheduler.log_match("Number of job equivalence classes: 1",
+                                 max_attempts=10, starttime=self.t)
+
+    def test_suspended_job_node(self):
+        "Test that a suspended job goes back to the same node on re queue"
```

Comment:
removed test 'test_suspended_job_node', since it was not doing anything different than what is already being tested.